lib/Controller/SettingsController.php

```diff
@@ -46,47 +72,11 @@ public function list() {
 	}
 
 	/**
-	 * FIXME: remove ./settings.php and this function, implement with OCP\Settings\ISettings instead
+	 * @AdminRequired
 	 *
-	 * @NoCSRFRequired
+	 * @return DataResponse
 	 */
-	public function settingsIndex() {
-		return new TemplateResponse(
-			$this->appName,
-			'settings',
-			['blank']
-		);
-	}
-
-	/**
-	 * FIXME: remove ./admin.php and this function, implement with OCP\Settings\ISettings instead
-	 */
-	public function adminIndex() {
-		return new TemplateResponse(
-			$this->appName,
-			'admin',
-			[
-				'wopi_url' => $this->appConfig->getAppValue('wopi_url'),
-				'edit_groups' => $this->appConfig->getAppValue('edit_groups'),
-				'doc_format' => $this->appConfig->getAppValue('doc_format'),
-				'test_wopi_url' => $this->appConfig->getAppValue('test_wopi_url'),
-				'test_server_groups' => $this->appConfig->getAppValue('test_server_groups'),
-				'canonical_webroot' => $this->appConfig->getAppValue('canonical_webroot'),
-				'menu_option' => $this->appConfig->getAppValue('menu_option'),
-				'encryption_enabled' => $this->appConfig->encryptionEnabled() ? 'true' : 'false',
-				'masterkey_encryption_enabled' => $this->appConfig->masterEncryptionEnabled() ? 'true' : 'false',
-				'secure_view_allowed' => $this->appConfig->enterpriseFeaturesEnabled() ? 'true' : 'false',
-				'secure_view_option' => ($this->appConfig->secureViewOptionEnabled() && $this->appConfig->enterpriseFeaturesEnabled()) ? 'true' : 'false',
-				'secure_view_open_action_default' => $this->appConfig->secureViewOpenActionDefaultEnabled() ? 'true' : 'false',
-				'secure_view_has_watermark_default' => $this->appConfig->secureViewHasWatermarkDefaultEnabled() ? 'true' : 'false',
-				'secure_view_can_print_default' => $this->appConfig->secureViewCanPrintDefaultEnabled() ? 'true' : 'false',
-				'watermark_text' => $this->appConfig->getAppValue('watermark_text')
-			],
-			'blank'
-		);
-	}
-
-	public function update($wopi_url, $edit_groups, $doc_format, $test_wopi_url, $test_server_groups, $external_apps, $canonical_webroot, $menu_option, $secure_view_option, $secure_view_open_action_default, $secure_view_can_print_default, $secure_view_has_watermark_default, $watermark_text) {
+	public function setAdminSettings($wopi_url, $edit_groups, $doc_format, $test_wopi_url, $test_server_groups, $external_apps, $canonical_webroot, $menu_option, $secure_view_option, $secure_view_open_action_default, $secure_view_can_print_default, $secure_view_has_watermark_default, $watermark_text, $zotero) {
```

Comment:
definitely we need a different approach. Sending 14 different parameters seems terrible, and the next time we need an additional parameter I'm sure we'll send 15 (as you've done already)

In general, I'd prefer to use an array and document what keys are required and what are optional.

Reply:
done, makes sense